doc/source/data/exchanging-datasets.rst

```diff
@@ -0,0 +1,50 @@
+.. _exchanging_datasets:
+
+===================
+Exchanging Datasets
```

Comment:
I feel like we need a better name for this. Maybe "Consuming Datasets"?

Reply:
I'm trying to understand what is this really about:
- the first example is about accessing dataset: this is a new thing - it's not creating/saving, it's not transforming (pipelining might be seen as an special/optimized case of transforming).
- the second one is actually transforming (one to many splitting).

Does it make sense to create a new page for Accessing Dataset, and fold the second case (splitting dataset) into transforming datasets?

Reply:
If we look at it in a more abstract way, we can categorize all operations into:
- Creating: File -> Dataset
- Saving: Dataset -> File
- Transforming: Dataset | List[Dataset] -> Dataset | List[Dataset]
- Accessing: Dataset[T] -> T | List[T] | Dict[K, V] ...

Notable features on top of those primitives:
- Pipelining: improved transforming. It's significant so it's worth separating out.
- Random accessing: it's essentially an accessor (although it's named as RandomAccessDataset, but it's not a Dataset)

And then we can organize documentation around them. WDYT?

Reply:
Great categorization! I like the "Accessing" page/category, my question is then around scope of that page:
- [x] Direct consumption: `.iter_rows()`, `.iter_batches()`, `.take()`, `.show()`, etc.
- [x] Exchange with ML frameworks: `.to_torch()`, `.to_tf()`. This can be seen as allowing an external framework to access/consume the dataset, and is ultimately a light wrapper around the direct consumption APIs (`.iter_batches()` in particular).
- [x] Random accessing: `RandomAccessDataset`. I'm thinking that this should be in "Accessing" as well, actually, since it's fundamentally providing random access to the underlying dataset, and the "Accessing" section is where I'd expect to find that functionality documented if I was a user.
- [ ] Exchange with other task frameworks: `.to_dask()`, `.to_spark()`, etc. This is currently under "Saving Datasets", which I still think makes the most sense.

Reply:
The scope looks good. For those to_X() methods, in theory they are similar to saving (an analogy came to mind is Google Doc can be printed to other file format like pdf). If it feels confusing for users (to be under Saving Datasets, or to be split across Saving and Accessing Datasets), we may add a Converting Datasets category which covers all those to_X()s that are used for inter-op (ML or data processing frameworks).